src/Features/Core/Portable/RemoveUnusedMembers/AbstractRemoveUnusedMembersDiagnosticAnalyzer.cs

```diff
@@ -0,0 +1,337 @@
+﻿// Copyright (c) Microsoft.  All Rights Reserved.  Licensed under the Apache License, Version 2.0.  See License.txt in the project root for license information.
+
+using System;
+using System.Collections.Generic;
+using System.Collections.Immutable;
+using System.Diagnostics;
+using System.Linq;
+using Microsoft.CodeAnalysis.CodeStyle;
+using Microsoft.CodeAnalysis.Diagnostics;
+using Microsoft.CodeAnalysis.Operations;
+using Microsoft.CodeAnalysis.Shared.Extensions;
+
+namespace Microsoft.CodeAnalysis.RemoveUnusedMembers
+{
+    internal abstract class AbstractRemoveUnusedMembersDiagnosticAnalyzer<TDocumentationCommentTriviaSyntax, TIdentifierNameSyntax>
+        : AbstractCodeStyleDiagnosticAnalyzer
+        where TDocumentationCommentTriviaSyntax: SyntaxNode
+        where TIdentifierNameSyntax : SyntaxNode
+    {
+        protected AbstractRemoveUnusedMembersDiagnosticAnalyzer(bool forceEnableRules)
+            : base(CreateDescriptors(forceEnableRules))
+        {
+        }
+
+        private static ImmutableArray<DiagnosticDescriptor> CreateDescriptors(bool forceEnableRules)
+        {
+            // TODO: Enable these rules by default once we have designed the Tools|Option location and UI for such code quality rules.
```

Comment:
note: my preference would be for this to actually be done with normal Options, and not this boolean parameter.  We do *not* have to make that option an editorconfig option.  *Nor* do we need to expose it in tools|options.

That will clean up this code, and make it so the tests don't need to pass this weird data along. Instead, they will just configure their options, like how every test normally works.